command/agent/agent_test.go

```diff
@@ -1291,6 +1291,48 @@ func TestServer_ShouldReload_ShouldHandleMultipleChanges(t *testing.T) {
 	}
 }
 
+func TestServer_ShouldReload_ReturnTrueForRPCUpgradeModeChanges(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
```

Comment:
Just fyi we're trying to get away from this package-shadowing naming style. Not a blocker.